code/modules/shuttle/on_move.dm

```diff
@@ -91,6 +91,8 @@ All ShuttleMove procs go here
 			return
 
 	loc = newT
+	if (newT.z != oldT.z && !istype(moving_dock, /obj/docking_port/mobile/supply)) // Cargo shuttles cannot carry living mobs which are, for now, the only things actually interested in Z transitions
+		onTransitZ(oldT.z, newT.z)
```

Comment:
Anything not crucial to the movement should be done in an after or before proc. The onMove procs are the only part of the move that isn't check ticked. This means the registered z will be inaccurate for a short time but that's fine.